operator/controllers/seldondeployment_controller.go

```diff
@@ -1438,16 +1438,16 @@ func jsonEquals(a, b interface{}) (bool, error) {
 	return bytes.Equal(b1, b2), nil
 }
 
-// Create Deployments specified in components.
-func (r *SeldonDeploymentReconciler) createDeployments(components *components, instance *machinelearningv1.SeldonDeployment, log logr.Logger) (bool, error) {
+// Create Deployments specified in components, returns ready, progressing, error.
+func (r *SeldonDeploymentReconciler) createDeployments(components *components, instance *machinelearningv1.SeldonDeployment, log logr.Logger) (bool, bool, error) {
```

Comment:
We can safely return true for progressing in most cases, we're only interested when the deployment is in a "bad state".